tools/donate_cpu_lib.py

```diff
@@ -42,12 +42,19 @@ def get_cppcheck(cppcheck_path, work_path):
             try:
                 os.chdir(cppcheck_path)
                 try:
+                    subprocess.check_call(['git', 'fetch'])
+                    localrev = subprocess.check_output(['git', 'rev-parse', 'main'])
+                    origrev = subprocess.check_output(['git', 'rev-parse', 'origin/main'])
+                    if localrev == origrev:
+                        print('Already up to date. Continuing with %s.' % localrev.decode().rstrip())
+                        time.sleep(2)
```

Comment:
I really think we should get rid of these sleeps since they make no sense and just slow things down. If results are being send there already is a 5 second sleep at the end of the analysis. And if no results are being send we still need to fetch the sources which should be enough of a delay.

Reply:
Will remove both time.sleep(2) calls as requested. 

I put time.sleep(2) before turn to be consistent with the existing "return True" statement, but will gladly remove them.